gravpm.c

```diff
@@ -265,6 +294,86 @@ static double sinc_unnormed(double x) {
     }
 }
 
+/* Compute neutrino power spectrum.
+ * This should happen after the CFT is computed,
+ * and after powerspectrum_compute() has been called,
+ * but before potential_transfer is called.*/
+static void compute_neutrino_power() {
+    if(!All.MassiveNuLinRespOn)
+        return;
+    /*Note the power spectrum is now in Mpc units*/
+    powerspectrum_sum(&PowerSpectrum, All.BoxSize*All.UnitLength_in_cm);
+    int i;
+    /*Get delta_cdm_curr , which is P(k)^1/2, and skip bins with zero modes:*/
+    int nk_nonzero = 0;
+    for(i=0;i<PowerSpectrum.size;i++){
+        if (PowerSpectrum.Nmodes[i] == 0)
```

Comment:
I think we shall eliminate these Nmodes[i] == 0 modes in the powerspectrum code -- here is where the linear binning works better -- at least all modes are non-empty!

Reply:
The linear binning actually produces a noticeably less accurate neutrino power spectrum, because of the excess smoothing induced (I checked this).

Reply:
excessive smoothing at low k? because log binning have excessive smoothing at high k.

I think what you really want is, probably something per mode at low-k but then averaged at high k.

Reply:
Excessive smoothing at low k, indeed. Excessive smoothing at high k is ok for the neutrinos (even physical!) I think what you suggest is exactly what I really want.

Reply:
We've been talking about doing this as an improvement of the model you implemented here. (actually a student here was supposed to work on this). Perhaps getting him to look at this after it is merged?

Reply:
That would be great.

Reply:
Ok. I'll start nudging him. It is relatively simple but will likely be a mixture of commits on K-space-neutrino and MP-Gadget as I see now.